buildkite/resource_pipeline_team_test.go

```diff
@@ -0,0 +1,226 @@
+package buildkite
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// Confirm we can add and remove a team pipeline resource with the default access level
+func TestAccPipelineTeam_AddRemoveWithDefaultsAccess(t *testing.T) {
+	var tp pipelineTeamResourceModel
+	teamName := acctest.RandString(12)
+	t.Parallel()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		CheckDestroy:             testCheckPipelineTeamResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config:             testAccPipelineTeamConfigBasic(teamName, "READ_ONLY"),
+				ExpectNonEmptyPlan: true,
```

Comment:
Why does this test expect a non-empty plan? I would expect this test case to create the 3 resources and then not have any other changes to make after that

Reply:
After running the tests, a terraform refresh is run, and this generates a non empty plan because the pipeline resource gets updated with the team block

Reply:
Yeah thats right, but thats not what we want. We want the pipeline.team block to ignore the appearance of new teams